src/Symfony/Component/Yaml/Inline.php

```diff
@@ -104,6 +108,18 @@ public static function dump($value, $exceptionOnInvalidType = false, $objectSupp
 
                 return 'null';
             case is_object($value):
+                if ($value instanceof \DateTime || $value instanceof \DateTimeImmutable) {
+                    if ($value->getTimezone()->getName() === date_default_timezone_get()) {
```

Comment:
What if it was explicitly defined as UTC and the server default timezone is also UTC ? You have no way to know whether the UTC timezone was explicitly chosen or no. But even worse, you have no idea how this will be parsed. So IMO, you should not drop information when dumping, assuming that the server parsing your YAML will have the same config